ggml-phi-knc.c

```diff
@@ -0,0 +1,142 @@
+/* Xeon PHI IMCI support. */
+/* formatted by using emacs, with (M-x set-variable RET c-basic-offset RET 4 RET) executed. */
+
+#include <stdint.h>
+
+// For size_t
+#include <stdio.h>
+
+// For memcpy.
+#include <string.h>
+
+// We can fit 16 of these float32s in a single vector register.
+#define GGML_F32_EPR 16
+
+// A vector of 16 floats.
+typedef float float32x16_t __attribute__((vector_size (64), aligned (64)));
+
+// A forward declaration, to keep GCC happy...
+void ggml_vec_dot_f32(int n, float * restrict s, size_t bs, const float * restrict x, size_t bx, const float * restrict y, size_t by, int nrc);
+
+inline static void GGML_F32x16_VEC_ZERO(float32x16_t *target)
+{
+    uint8_t zero[4] __attribute__((aligned(64))) = {0,0,0,0};
+
+    __asm__ __volatile__ (
+                          "vbroadcastf32x4\t%[Z]%{uint8%},\t%%zmm8\n\t" // use an upscaling operator to clear our value.
+                          "vmovnraps\t\t%%zmm8,\t%[RES]\n\t"
+                          : [RES]  "+m"  (*target)
+                          : [Z]    "m"   (zero)
+                          : "zmm8");
+
+}
+
+// Multiply each item in mvec1 with the corresponding item in mvec2, adding the result to the corresponding item in sum. optionally clear the sum before starting. 
```

Comment:
Remove trailing whitespace:
```suggestion
// Multiply each item in mvec1 with the corresponding item in mvec2, adding the result to the corresponding item in sum. optionally clear the sum before starting.
```